lib/screens/v2/candidate_generator/on_bus/departures.ex

```diff
@@ -80,6 +82,47 @@ defmodule Screens.V2.CandidateGenerator.Widgets.OnBus.Departures do
     Enum.filter(departures, &(&1.prediction.route.id != route_id))
   end
 
+  defp filter_duplicate_routes(departures) do
+    unique_departures =
+      Enum.uniq_by(departures, fn dep ->
+        {dep.prediction.route.id, dep.prediction.trip.direction_id}
+      end)
+
+    # Return at least 3 departures if they exist, even if there are not enough unique routes
+    case length(unique_departures) do
+      count when count >= 3 ->
+        unique_departures
+
+      _ ->
+        unique_departures ++
+          Enum.take(departures -- unique_departures, 3 - length(unique_departures))
+    end
+  end
+
+  defp sort_by_mode(departures) do
+    priority = %{
+      :subway => 1,
+      :light_rail => 1,
+      :bus => 2,
+      :commuter_rail => 3,
+      :ferry => 4
+    }
+
+    Enum.sort(departures, fn a, b ->
+      priority_a =
+        Map.get(priority, a.prediction.route.type)
+
+      priority_b =
+        Map.get(priority, b.prediction.route.type)
+
+      if priority_a == priority_b do
+        a.prediction.arrival_time <= b.prediction.arrival_time
```

Comment:
Departures should already be sorted the way we want coming from the API, so ideally we wouldn't duplicate that here. Elixir sort functions are stable, so they should leave departures in the correct order within their priority groups after sorting by priority. (It seems like you might have added this to compensate for `filter_duplicate_routes` not necessarily leaving departures in the order it found them, though see my comment there as well. In case we do want to keep that behavior, I think it should be possible for that function to be "stable" with regard to ordering as well.)

Reply:
Good point, I can definitely keep that `filter_duplicate_routes` behavior and maintain ordering by time as well so that this comparison of time isn't necessary.